pkg/metrics/http/server_metrics.go

```diff
@@ -119,16 +124,35 @@ func (m *ServerMetrics) InstrumentationHandler(next http.Handler) http.Handler {
 		next.ServeHTTP(wd, r)
 
 		method := r.Method
-		code := wd.Status()
+		statusCode := wd.Status()
 		handlerName := mux.CurrentRoute(r).GetName()
 
-		m.serverReqTotalCounter.WithLabelValues(handlerName, method, code).Inc()
+		// get error response code if exists
+		var (
+			errorResp = &modelgateway.ErrorResponse{}
+			errorCode string
+		)
+		if statusCode != strconv.Itoa(http.StatusOK) {
+			body := wd.GetBody()
+			err := xml.Unmarshal(body, errorResp)
+			if err != nil {
+				log.Errorw("cannot parse gateway error response", "error", err)
+				errorCode = "-1" // unknown error code
+				goto METRICS
+			}
+			errorCode = strconv.Itoa(int(errorResp.Code))
+		} else {
+			errorCode = "0" // no error
+		}
```

Comment:
These codes can be encapsulated in wd.GetIntervalErrorCode?

Reply:
refactored